src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs

```diff
@@ -123,7 +124,36 @@ public override bool Execute()
                 if (string.IsNullOrEmpty(request.NuspecFile))
                 {
                     packageBuilder = logic.GetPackageBuilder(request);
+                    string noWarnProperties = null;
+
+                    if (packageBuilder?.Properties?.TryGetValue("NoWarn", out noWarnProperties) == true
+                        && !string.IsNullOrWhiteSpace(noWarnProperties))
+                    {
+                        PackageSpecificWarningProperties packageSpecificWarningProperties = new();
+
+                        foreach (string packageNoWarnProperty in noWarnProperties.Split(new string[] { Environment.NewLine },
+                            StringSplitOptions.RemoveEmptyEntries))
+                        {
+                            string[] packageNoWarnPropertyPars = packageNoWarnProperty.Split(';');
+                            string nugetFramework = packageNoWarnPropertyPars[0];
+                            string packageId = packageNoWarnPropertyPars[1];
+
+                            foreach (string noWarnProperty in packageNoWarnPropertyPars[2].Split(' '))
+                            {
+                                packageSpecificWarningProperties.Add((NuGetLogCode)Enum.Parse(typeof(NuGetLogCode), noWarnProperty), packageId, NuGetFramework.Parse(nugetFramework));
```

Comment:
Should we call `TryParse()`, or is it impossible for this list to have something that's not part of the `NuGetLogCode` enum?

Reply:
We should totally call Try Parse. 

there's no guarantee that pack and restore are run with the same tooling and also someone specifying a code that doesn;'t exist should fail their whole scenario.